modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -0,0 +1,1016 @@
+                            std::stringstream ss;
+                            ss << "OV Backend: Provided tensor layout: " << *explicit_in_tensor_layout
+                               << " is not compatible with input data: " << matdesc
+                               << ". Expecting: NHWC";
+                            util::throw_error(std::logic_error(ss.str()));
+                        }
+                    }
+                    input_info.tensor().set_layout(::ov::Layout("NHWC"));
+                    input_info.tensor().set_spatial_static_shape(matdesc.size.height,
+                                                                 matdesc.size.width);
+                    // NB: Even though resize is automatically configured
+                    // user have an opportunity to specify the interpolation algorithm.
+                    auto interp = explicit_resize
+                        ? toOVInterp(*explicit_resize)
+                        : ::ov::preprocess::ResizeAlgorithm::RESIZE_LINEAR;
+                    input_info.preprocess().resize(interp);
+                } else {
+                    // NB: Othersize configure resize only if user asked.
+                    if (explicit_resize) {
+                        if (matdesc.isND()) {
+                            // NB: ND case - need to obtain "H" and "W" positions
+                            // in order to configure resize.
+                            if (!explicit_in_tensor_layout && model_layout.empty()) {
+                                std::stringstream ss;
+                                ss << "Resize for input layer: " << input_name
+                                   << "can't be configured."
+                                   << " Failed to extract H and W positions from layout.";
+                                util::throw_error(std::logic_error(ss.str()));
+                            } else {
+                                const auto layout = explicit_in_tensor_layout
+                                    ? ::ov::Layout(*explicit_in_tensor_layout) : model_layout;
+                                const auto H_idx = ::ov::layout::height_idx(layout);
+                                const auto W_idx = ::ov::layout::width_idx(layout);
+                                // NB: It still possible if layout is "??HW".
+                                GAPI_Assert(H_idx >= 0 && H_idx < static_cast<int>(matdesc.dims.size()));
+                                GAPI_Assert(W_idx >= 0 && W_idx < static_cast<int>(matdesc.dims.size()));
+                                input_info.tensor().set_spatial_static_shape(matdesc.dims[H_idx],
+                                                                             matdesc.dims[W_idx]);
+                                input_info.preprocess().resize(toOVInterp(*explicit_resize));
+                            }
+                        } else {
+                            // NB: 2D case - know exactly where H and W...
```

Comment:
Could you please add some debug messages here other that throws? I mean what we assume in which case

Reply:
Here = part with image/tensor logic. Messages = at least comments you put here